test/test_file_digest.py

```diff
@@ -46,14 +57,85 @@ def test_description_message(tmpdir, digestcheck):
     assert output.get_description('cron-file-digest-unauthorized') == 'Please refer to\nhttps://en.opensuse.org/openSUSE:Package_security_guidelines#audit_bugs for\nmore information.\n\n'
 
 
-def test_from_simple_dummy_pkg(digestcheck):
-    output, test = digestcheck
-    with FakePkg('dummy') as pkg:
-        pkg.add_file_with_content('/etc/polkit-1/rules.d/r.txt', 'Hello world')
-        pkg.add_file_with_content('/root/sample.txt', 'Hello world')
-        pkg.add_symlink_to('/etc/polkit-1/rules.d/r2.txt', '../../root/sample.txt')
+def test_matching_digests():
+    output, test = get_digestcheck('digests.config')
+    with FakePkg('testpkg') as pkg:
+        pkg.add_file_with_content('/restricted/1/dangerous', 'really dangerous stuff')
+        # also test following of symlinks with this entry
+        pkg.add_symlink_to('/alsorestricted/2/suspicious', '/other/place/suspicious.txt')
+        pkg.add_file_with_content('/other/place/suspicious.txt', 'really suspicious stuff')
+        pkg.add_file_with_content('/related/and/also/sensitive', 'related sensitive stuff')
+        test.check(pkg)
+        assert len(output.results) == 0
+
+
+def test_simple_mismatch():
+    output, test = get_digestcheck('digests.config')
+    with FakePkg('testpkg') as pkg:
+        pkg.add_file_with_content('/restricted/1/dangerous', 'really dangerous stuff')
+        pkg.add_symlink_to('/alsorestricted/2/suspicious', '/other/place/suspicious.txt')
+        pkg.add_file_with_content('/other/place/suspicious.txt', 'really good stuff')
+        pkg.add_file_with_content('/related/and/also/sensitive', 'related sensitive stuff')
+        test.check(pkg)
+        assert len(output.results) == 1
+        if output.results:
+            error = output.results[0]
```

Comment:
You can make this check unconditional.

Reply:
I actually deliberately kept the `if` condition, because if the prior assertion fails and output.results is empty then an unhelpful follow-up error (IndexError) will occur. If you think it is better without that then I can adjust, of course.

Reply:
I thought that I saw in my test runs that the initial assertion does not abort execution of the test. But maybe I am wrong about that.

Reply:
That should not happen, I tested that locally :)